### PR TITLE
DDFLSBP-593/add azure communication service mailer module

### PR DESCRIPTION
**Pull request by Dresse**
#### Link to issue

https://reload.atlassian.net/browse/DDFLSBP-593
https://reload.atlassian.net/browse/DDFLSBP-572

#### Description
DDFLSBP-593:
This PR adds functionality for sending out emails using Azure Communication Services Mailer by enabling the azure_mailer and mailsystem modules. The azure_mailer module is configured using environment variables as the endpoint and secret fields. This is done in all.settings.php. 
We also configure the rest of the site to use azure_mailer module by changing the site email address and setting up the mail system to use Azure Communication Service as the sites default formatter and sender. 

We also update the contact webform email handlers with new custom message bodies for both confirmation and notification emails. Furthermore we add danish translations to all of the used form fields and submission texts.

DDFLSBP-572:
The PR moves the webform related hooks into the correct module "dpl_webform_configuration".
We also updated the way the notification email handler works. Now it will use email associated with the selected term in the webform submission. This will make it possible to send email depending on what category is chosen.

#### Additional comments or questions

We need to add `'AZURE_MAILER_SECRET'` and `'AZURE_MAILER_ENDPOINT'` as environment variables. 
We do this manually on our demo site, but Deranged has a task for setting it up on all environments going forward.

### CODE REVIEW

File: config/sync/system.site.yml [hunk 1]
```diff
@@ -3,11 +3,11 @@ _core:
 langcode: en
 uuid: 13ef1a53-dfb4-4c82-9b64-44586a366729
 name: 'DPL CMS'
-mail: admin@example.com
+mail: DoNotReply@folkebibliotekernescms.dk
 slogan: ''
 page:
   403: ''
-  404: /customerror/404
+  404: ''
```
Comment (rasben) [inline config/sync/system.site.yml:10]: Is this change on purpose?

File: config/sync/webform.webform.contact.yml [hunk 2]
```diff
@@ -155,7 +155,7 @@ settings:
   confirmation_type: message
   confirmation_url: '<front>'
   confirmation_title: ''
-  confirmation_message: '<p>Din besked er blevet sendt.</p>'
+  confirmation_message: '<p>Din forespørgsel er blevet modtaget. Du vil modtage en bekræftelse på mail.</p>'
```
Comment (rasben) [inline config/sync/webform.webform.contact.yml:158]: I think English should be the standard and Danish translations is the job of DDF to manage

File: config/sync/webform.webform.contact.yml [hunk 3]
```diff
@@ -251,13 +251,13 @@ handlers:
...
       from_options: {  }
-      from_name: _default
+      from_name: ''
       reply_to: ''
       return_path: ''
       sender_mail: ''
       sender_name: ''
       subject: '[webform_submission:values:subject:raw]'
-      body: '[webform_submission:values:message:value]'
+      body: '<p>Hej [webform_submission:values:name]&nbsp;</p><p>Vi bekræfter hermed at vi har modtaget følgende forespørgsel:&nbsp;</p><p>Modtaget: [webform_submission:created]</p><p>Forespørgsel:<br>[webform_submission:values:subject]</p><p>[webform_submission:values:message]</p><p>---------</p><p>Vi vil besvare dig hurtigst muligt på følgende email: [webform_submission:values:email]&nbsp;</p>'
```
Comment (rasben) [inline config/sync/webform.webform.contact.yml:260]: I think English should be the standard and Danish translations is the job of DDF to manage

File: config/sync/webform.webform.contact.yml [hunk 4]
```diff
@@ -280,21 +280,21 @@ handlers:
...
       from_options: {  }
-      from_name: '[webform_submission:values:name:raw]'
+      from_name: ''
       reply_to: ''
       return_path: ''
       sender_mail: ''
       sender_name: ''
       subject: '[webform_submission:values:subject:raw]'
-      body: '[webform_submission:values:message:value]'
+      body: '<p>Der er kommet en ny forespørgsel med følgende info:</p><p>Navn: [webform_submission:values:name]&nbsp;</p><p>Email: [webform_submission:values:email]&nbsp;</p><p>Modtaget: [webform_submission:created]</p><p>Kategori: [webform_submission:values:category]</p><p>Emne: [webform_submission:values:subject]</p><p>Besked: [webform_submission:values:message]</p>'
```
Comment (rasben) [inline config/sync/webform.webform.contact.yml:297]: I think English should be the standard and Danish translations is the job of DDF to manage

File: web/modules/custom/dpl_webform_configuration/dpl_webform_configuration.module [hunk 5]
```diff
@@ -15,3 +15,29 @@ function dpl_webform_configuration_webform_create(Webform $webform): void {
   // Therefore we disable saving of results for all new webforms.
   $webform->setSetting('results_disabled', TRUE);
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function dpl_webform_configuration_preprocess_input__submit(array &$variables): void {
+  $original_classes = $variables['attributes']['class'];
```
Comment (rasben) [inline web/modules/custom/dpl_webform_configuration/dpl_webform_configuration.module:23]:
```suggestion
  $original_classes = $variables['attributes']['class'] ?? [];
```

File: web/themes/custom/novel/novel.theme [hunk 6]
```diff
@@ -326,29 +326,3 @@ function novel_preprocess_views_view_unformatted(&$variables): void {
     }
   }
 }
-
```
Comment (rasben) [inline web/themes/custom/novel/novel.theme:329]: Awesome that you move functionality away from the theme :)

File: config/sync/system.site.yml (hunk 1 above)
Comment (Dresse) [reply config/sync/system.site.yml:10]: Yes. We have removed the customerror module and thus the page no longer exist. We just forgot to remove it here.

File: config/sync/webform.webform.contact.yml (hunk 2 above)
Comment (Dresse) [reply config/sync/webform.webform.contact.yml:158]: I think you are somewhat correct. I myself wasn't sure how to handle this.
The reason I ended up doing it like this is because DDF going forward are supposed to be able to create their own webforms and probably are gonna write these things in danish instead of first writing them in english and then translating.
But I'll change it to english for now.

File: config/sync/webform.webform.contact.yml (hunk 4 above)
Comment (Dresse) [reply config/sync/webform.webform.contact.yml:297]: I think you are somewhat correct. I myself wasn't sure how to handle this.
The reason I ended up doing it like this is because DDF going forward are supposed to be able to create their own webforms and probably are gonna write these things in danish instead of first writing them in english and then translating.
But I'll change it to english for now.

**Review by spaceo — APPROVED** [review]
LGTM 👍

File: web/themes/custom/novel/novel.theme (hunk 6 above)
Comment (Dresse) [reply web/themes/custom/novel/novel.theme:329]: Thanks 😁
